server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTask.java

```diff
@@ -203,26 +219,67 @@ private DatabaseChangeThread generateOdcAsyncTaskThread(Long taskId, DelegateExe
     }
 
     private void modifyTimeoutIfTimeConsumingSqlExists(DelegateExecution execution,
-            DatabaseChangeParameters parameters) {
-        if (!parameters.isModifyTimeoutIfTimeConsumingSqlExists()) {
+            DatabaseChangeParameters parameters, DialectType dialectType, Long creatorId) {
+        long autoModifiedTimeout = flowTaskProperties.getIndexChangeMaxTimeoutMillisecond();
+        if (!parameters.isModifyTimeoutIfTimeConsumingSqlExists() || !dialectType.isOceanbase()
+                || autoModifiedTimeout <= parameters.getTimeoutMillis()) {
             return;
         }
-        Long taskId = FlowTaskUtil.getTaskId(execution);
-        Long preCheckTaskId = FlowTaskUtil.getPreCheckTaskId(execution);
-        TaskEntity preCheckTask = taskService.detail(preCheckTaskId);
-        PreCheckTaskResult preCheckResult =
-                JsonUtils.fromJson(preCheckTask.getResultJson(), PreCheckTaskResult.class);
-        Validate.notNull(preCheckResult, "Pre check task result can not be null");
-        long autoModifiedTimeout = flowTaskProperties.getIndexChangeMaxTimeoutMillisecond();
-        if (Objects.nonNull(preCheckResult.getSqlCheckResult())
-                && preCheckResult.getSqlCheckResult().isTimeConsumingSqlExists()
-                && autoModifiedTimeout > parameters.getTimeoutMillis()) {
-            this.autoModifyTimeout = true;
-            parameters.setTimeoutMillis(autoModifiedTimeout);
-            TaskEntity databaseChangeTaskEntity = taskService.detail(taskId);
-            databaseChangeTaskEntity.setParametersJson(JsonUtils.toJson(parameters));
-            taskService.updateParametersJson(databaseChangeTaskEntity);
+        List<OffsetString> userInputSqls = null;
+        SqlStatementIterator uploadFileSqlIterator = null;
+        String delimiter = parameters.getDelimiter();
+        if (StringUtils.isNotBlank(parameters.getSqlContent())) {
+            userInputSqls = SqlUtils.splitWithOffset(dialectType, parameters.getSqlContent(), delimiter);
+        }
+        if (CollectionUtils.isNotEmpty(parameters.getSqlObjectIds())) {
+            String bucketName = "async".concat(File.separator).concat(creatorId.toString());
+            InputStream uploadFileInputStream =
+                    databaseChangeFileReader.readInputStreamFromSqlObjects(parameters, bucketName, -1);
+            if (uploadFileInputStream != null) {
+                uploadFileSqlIterator =
+                        SqlUtils.iterator(dialectType, delimiter, uploadFileInputStream, StandardCharsets.UTF_8);
+            }
+        }
+        while (CollectionUtils.isNotEmpty(userInputSqls)
+                || (uploadFileSqlIterator != null && uploadFileSqlIterator.hasNext())) {
+            String sql = CollectionUtils.isNotEmpty(userInputSqls) ? userInputSqls.remove(0).getStr()
+                    : uploadFileSqlIterator.next().getStr();
```

Comment:
remove element in iterator in loop  is unsafe operation,  suggest  use iterator

Reply:
The object being iterated and the object being revomed are not the same